### ISSUE TITLE
Unable to customize. File 'D:/config.txt' does not existed

### ISSUE THREAD

**Issue by FrankJian** [raspberrypi/rpi-imager#289]
I try use the imager (v1.6.2) to write the image to SD card (16GB) with customize configuration (Ctrl + Shift + X) to config the SSH and set locale , writing success but failed in verif step , it show the error "Unable to customize. File 'D:/config.txt' does not existed"
i use the win10 to do it .
step 1:
![image](https://user-images.githubusercontent.com/24399952/137567560-80aad539-d650-42d5-9dd4-9f59bcfd41b8.png)
Step 2:
![image](https://user-images.githubusercontent.com/24399952/137567585-fabf1e2d-17c4-4df9-aa22-d6d40e1b300f.png)

Step 3:
![image](https://user-images.githubusercontent.com/24399952/137567606-9fdeea1d-d079-4123-aa1c-3ff61417fd2e.png)

And the result:
![image](https://user-images.githubusercontent.com/24399952/137567538-2a88566b-e225-4da7-b1b1-9b2117e8c740.png)

**Comment by MorganMarshall** [comment]
I'm also having the same issue.

**Comment by tjapro** [comment]
As a comment from linux user:

I've also made some "Advanced Settings" and the error that it gives is:

`The file config.txt do not exist at /run/media/user/boot/` (more or less)

But when I'm going to check it at file manager (as `root`) there is the config.txt, but there is also the `/boot` partition from the pc itself.

I think that, as the program uses escalated privileges (`root`), it only check one partition, mount point, or drive.

Program Version: flatpak 1.6.2
OS Version: Fedora 35

**Comment by maxnet** [comment]
>Program Version: flatpak 1.6.2

Does it work better with the .rpm the Fedora guys created?

**Comment by milanbrahmbhatt** [comment]
Just uncheck the auto ejection option from the advanced settings, and things work fine. Looks like the bug is custom config.txt is being written after the auto-ejection

**Comment by otzelpov** [comment]
I also encountered this issue on my linux mint machine (flathub), unchecking the auto ejection option did not helped. On my mac it worked without any problems.

**Comment by IPlayZed** [comment]
Same here, using Arch, hardened kernel. The `config.txt` does exist tho. I still haven't checked if the customization options are actually set, can someone validate it?

**Comment by milanbrahmbhatt** [comment]
Custom options appear to be set when trying on Mac OS, my raspberry pi 4 model B successfully connects to WiFi on boot up.

**Comment by maxnet** [comment]
>Same here, using Arch

Natively compiled for Arch, or using one of the container formats like flatpak/snap/etc. ?

**Comment by IPlayZed** [comment]
> >Same here, using Arch
> 
> Natively compiled on Arch, or using one of the container formats like flatpak/snap/etc. ?

Flatpak.

**Comment by fishd72** [comment]
Having the same issue here on Linux Mint using the flatpak (installed via the Mint Software Manager), it shows this error on every SD card write.

I've just downloaded and installed the .deb direct from Raspberrypi.com and it seems to work fine. No errors and config written to the SD card successfully.

**Comment by lurch** [comment]
> I've just downloaded and installed the .deb direct from Raspberrypi.com and it seems to work fine.

That's good to hear :+1: 

> Having the same issue here on Linux Mint using the flatpak

Problems with the flatpak version should be reported at https://github.com/flathub/org.raspberrypi.rpi-imager/issues

**Comment by jhgarrison** [comment]
Same here, in addition, here's what the File Explorer looks like...

![rpi001](https://user-images.githubusercontent.com/2840152/150194838-214f99a2-ecb1-45e6-9aad-77931d059be9.png)

**Comment by maxnet** [comment]
>Same here, in addition, here's what the File Explorer looks like...

Is it the same if you let Imager download an image from our repository?

Judging by the image name starting with "202" I think you are using a custom image you downloaded outside of imager, in which case we cannot verify that the download went correctly. (Normally it does a checksum hash check).

**Comment by jhgarrison** [comment]
The formatting on the SD card was somehow funky.  I ended up having to nuke the first 512MB (Cygwin dd if=/dev/zero of=/dev/sdh...) of the card before Windows would consent to formatting it.  After that the imager write to the card proceeded about 3 times faster than the first time.

I skipped customization in the imager this time... I'll try later with customization to see if that causes the problem.

**Comment by andrum993** [comment]
> The formatting on the SD card was somehow funky. I ended up having to nuke the first 512MB (Cygwin dd if=/dev/zero of=/dev/sdh...) of the card before Windows would consent to formatting it. After that the imager write to the card proceeded about 3 times faster than the first time.
> 
> I skipped customization in the imager this time... I'll try later with customization to see if that causes the problem.

Something had corrupted the contents of the SD card - the directory listing shows garbage has ended up where the FAT root directory should be.

**Comment by sroller** [comment]
I switched back to v1.6.2 to make it work.

**Comment by BenediktCromind** [comment]
I was able to find a workaround. 
1. After the error message I went into the Windows Disk Management. The boot partition wasn't mounted.
2. I mounted the partition and assigned a drive letter.
3. I installed Raspberry PI OS again. This time the installation run through without the error.

Maybe that helps in fixing this. It seems that Windows sometimes doesn't assign a drive letter automatically to the boot partition.

**Comment by stefan-sherwood** [comment]
I have the same problem, writing to a MicroSD card using a new Samsung reader that came with it. I tried disabling "Eject" and the problem still happens. Removing the drive and putting it back in, I see that I am also getting corrupted file names:

![image](https://user-images.githubusercontent.com/2094779/159334778-e5821136-82a1-4602-a256-4797aac815b2.png)

I have used the RPi imager with this same equipment to install the same OS a bunch of times so I suspect there is some sort of race condition.

**Comment by stefan-sherwood** [comment]
I have the same problem, writing to a MicroSD card using a new Samsung reader that came with it. I tried disabling "Eject" and the problem still happens. Removing the drive and putting it back in, I see that I am also getting corrupted file names:

![image](https://user-images.githubusercontent.com/2094779/159335056-851c7b2b-3982-43e1-a0a1-5f6fa27cf899.png)

**Comment by tsjk** [comment]
I have the same problem using v1.7.2 from flathub on my Gentoo x86_64 system. The natively compiled version bails out due to some ```qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat``` problem. But that's another story.

**Comment by maxnet** [comment]
>The natively compiled version bails out due to some qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat problem.

If using nvidia make sure you have recent drivers.

**Comment by tsjk** [comment]
I don't. I read somewhere that someone figured out that recompiling some packages without glvnd makes it work. I got the imager to work on my macOS system, so I won't be recompiling anything.

**Comment by yisonPylkita** [comment]
I also had the same error message. I nuked first 2GBs of my SD card and unplugged it. After that I was able to successfully install system image. I did this on Windows 10 using Disk Manager (disable Quick Format)

**Comment by enzorpl** [comment]
What part of the memory did you do this? Because I can not format the part that should have the OS.

**Comment by ac0mputerguru** [comment]
> > 
> 
> Can you explain more about what did you do? You literally erased just the first 2 GBs of memory from 
> your SD card and it worked?

TLDR: Try being patient and see it that helps. Proper USB prep is also a key item.

I hope this helps. I have a 32 Gb USB memory stick that I'm trying to load the 64-bit Debian Lite (Bullseye) onto with a custom config. I kept getting the same "d:/config.txt" error. I have used Windows since version 2.0. Sometimes,  a problem will happen today and it reminds me of an old solution that's filed away in my cob-webby brain.

First rule of thumb, reformat the drive. I'm super cautious with diskmgmt, as it has specific rules it seems to follow. To reformat the drive, first delete all the partitions ON THAT DRIVE. You should end up with the entire drive being unallocated. Quit disk management and go back into it. This makes the OS rescan/refresh the internal drive tables.

Then, format the drive as an MBR->FAT32 drive. It should also have a "proper" FAT volume name: ALL CAPS, MAX 11 CHARACTERS, ONLY USE - AND _ AS PUNCTUATION. Personally, I just name mine MT (see what I did there? But that's honestly what I name FAT drives that are going to be erased anyway.) 

Now, exit disk management and then EJECT the disk, using the "Safely Remove Hardware and Eject Media" icon in the lower right corner, by the clock. Ejecting the disk forces Windows to close all the file handles and release the disk information from it's tables. Re-insert the drive and Windows will read the disk and update the drive tables, once again.

After prepping the USB drive, I used Raspberry Pi Imager 1.7.2 and chose options:  Raspberry Pi OS Other, 64-bit Bullseye Lite as my source. I selected "MT" and set my options. I turned off "Eject media when finished", then wrote to "MT".

(Get ready for the EUREKA!! moment...)

Windows popped up the "you must format or eject" dialog and I grabbed for the mouse. Like I always do. That's when the old sense memory hit me and I literally slapped my hand as I said, "Wait for it..." The Imager app stated it was waiting for disk D: to re-mount. After about 30-40 seconds, VOILA! It completed as it was supposed to! I test it a couple more times, just to ensure my notes and method was consistent.

Why does this work? I remember a LONG time ago, back in the XP days, that there was a problem with installing Windows because it couldn't find the CD media. Come to find out, the disk management/hot plug/notification system(s) can sometimes take a LONG time to message each other internally. 

After you'd format the hard drive, it would should up as the drive C you expected to see. However, as the internal drive tables were updated, the CD drive would automatically be re-assigned a different drive letter. Now, the installer freaks out because it can't find the source media. Annoying as hell, since SOMETIMES it would install if I simply walked away a left the machine for an hour or so. Finally, I figured it out and it made so much sense, in hindsight. 

I remember, during future installs, waiting for the error box to pop up and then slow counting to 60 before hitting the "Try Again" button. Worked like a charm. I just had to be patient enough to let Windows figure it's crap out on it's own. And their installation app programmers needed to learn that same patience, LOL

It's the same deal, here. Try being patient and see it that helps. Proper USB prep is also a key item.

**Comment by ac0mputerguru** [comment]
> Same here, in addition, here's what the File Explorer looks like...
> 
> ![rpi001](https://user-images.githubusercontent.com/2840152/150194838-214f99a2-ecb1-45e6-9aad-77931d059be9.png)

This looks like "classic failure to eject" disk corruption. Those funny dates (esp 1984 and 2038) were common when USB disks very first came out. Windows buffers USB devices, because if you have enough memory, you can cache the write-data and dweedle it out to the slower USB device. This allows Windows to move on to the next task you're asking it to do.

USB was heavily advertised as "plug-and-play" when first marketed in the USofA. Unfortunately, most people didn't know about the write-caching and would just yank the USB drive out and walk away. What Microsoft (quietly) told TechNet subscribers is "always, always, ALWAYS eject any USB storage before physically removing it, to avoid data corruption". 

A couple of years ago, I saw Rufus or Etcher suggest ejecting and then re-inserting the USB drive before attempting to install it on a Raspberry Pi. Kinda tickled my memory, lol.

**Comment by Guiorgy** [comment]
@ac0mputerguru So essentially you believe that waiting for ~60 seconds before pressing cancel should help Windows get it's s**t together and work? I'll definitely try it next time I flash Raspbian and give an update!

**Comment by ac0mputerguru** [comment]
I'm working on a touchy project and had to burn the USB image a couple of times today. I watched each time as the Imager app waited for the drive to remount, then it finished the job. I also noticed the format dialog talked about drive E:, not D:, like it used to. I think this is the solution for me.

**Comment by JaimeNoyola** [comment]
Hello everyone! 
I found the solution! You download ISO Raspbian in this page https://www.raspberrypi.com/software/operating-systems/ after you download balenaetcher and with this program you mount the ISO in your Micro SD and ready! 
Greetings to all!

**Comment by tomas-mraz** [comment]
I had the same problem and then ... I download the image (64bit lite) separately.
Windows 10, USB card reader on USB hub.
First I erased the card by app's built-in functionality.
After writing and verifying actions suddenly appears app's popup window with something like "waiting to mount filesystem" (I saw it first time) ... after 3 seconds disappeared and work was done! I used the card and it works fine with config settings.
From my point of view ... when an image is downloading on the fly, the app does not wait during the mounting of the card's FS.
Hope it helps.
PS: During the verification phase, the SD card looks unused in Windows Disk Manager.

**Comment by ac0mputerguru** [comment]
Lovely feedback, mate! Unfortunately, I'm old school enough to always download an image before I use it. On the fly is only if I'm desperate. So, I was always using a download as my source. This truly is all about timing and how Windows works. I'm on Win10 Pro with latest updates. Tried it again this morning and still working as I originally described, including my solution.

**Comment by Roei-Be** [comment]
> I'm also having the same issue. Edit: I just did the erase function, then retried and it worked.

tnx, worked for me too :)

**Comment by Ashcora** [comment]
> I'm also having the same issue. Edit: I just did the erase function, then retried and it worked.

I had the same issue. It also worked for me to erase the SD card first using the Windows partition manager.

**Comment by IntaekYeo** [comment]
> I was able to find a workaround.
> 
> 1. After the error message I went into the Windows Disk Management. The boot partition wasn't mounted.
> 2. I mounted the partition and assigned a drive letter.
> 3. I installed Raspberry PI OS again. This time the installation run through without the error.
> 
> Maybe that helps in fixing this. It seems that Windows sometimes doesn't assign a drive letter automatically to the boot partition.
I did it like you said and it worked out.

1. Problem Occurred
2. Execute "disk management" to delete divided partitions and merge all of them.
3. Reinstall with "imager"
4. No problem.

**Comment by tdewey-rpi** [comment]
Closing as 'Won't Fix'.

Thanks to the commenters in this thread, who have identified that this is a curious Windows-dependent behaviour. It's unfortunately not something we can address in Raspberry Pi Imager beyond some guidance - and in that case we don't have a useful way to deliver that in the current UI.

As they say, there are only two hard problems in Computer science: cache invalidation, naming things, and off-by-one errors.
